archive-tar.c

```diff
@@ -454,26 +455,36 @@ static int write_tar_filter_archive(const struct archiver *ar,
 	if (args->compression_level >= 0)
 		strbuf_addf(&cmd, " -%d", args->compression_level);
 
+	snprintf(outmode, sizeof(outmode), "%s%d", "wb", args->compression_level);
```

Comment:
`snprintf()` will no longer be allowed in Git for Windows' code base, as soon as v2.21.0 comes out. It is not even necessary here: you can write

```c
char outmode[4] = "wb\0";
if (args->compression_level >= 0 && args->compression_level <= 9)
    outmode[2] = '0' + args->compression_level;
```

BTW please move the declaration of `outmode` to the `if (!strcmp("gzip -cn", ar->data))` block, as that is the right scope for that variable.